### ISSUE TITLE
The type definition of highstock is incorrect

### ISSUE THREAD

**Issue by scott-ho** [highcharts/highcharts#9705]
#### Expected behaviour
import the highstock types correctly.

#### Actual behaviour
import with build errors

#### Live demo with steps to reproduce
Just install the v7 highcharts and drilldown to see `node_modules/highcharts/highstock`, the path is invalid

![image](https://user-images.githubusercontent.com/35443789/50137398-fd271f80-02d5-11e9-8302-d629ca339b5d.png)

#### Product version
v7.0

#### Affected browser(s)
ALL

**Comment by bre1470** [comment]
Thank you for the report! This will be indirectly resolved with https://github.com/highcharts/highcharts/pull/9692

**Comment by scott-ho** [comment]
@bre1470 how does the PR resolve the problem. The changes inside it doesn't involve related code.

**Comment by bre1470** [comment]
The fix is part of v1.0.5 declarations generator.

**Comment by KacperMadej** [comment]
@scott-ho 
If you need to load Highstock with current version in your project you could use a module.
**Workaround**: load stock as a module, so like:

```ts
import * as Highcharts from 'highcharts';
import StockModule from 'highcharts/modules/stock';
StockModule(Highcharts);
```

**Comment by scott-ho** [comment]
I'm afraid that it's not very feasible. We have many many imports from highstock.

BTW. When will it be released?

**Comment by bre1470** [comment]
Highcharts v7.0.1 will be released this week and fix the problem.

**Comment by gbenthomas** [comment]
It looks like even in v7.0.1, the path in highcharts/highstock.d.ts is wrong: 

`import StockModule from '../modules/stock';`

**Comment by KacperMadej** [comment]
@gbenthomas Thank you very much for finding this one out. Only that file has this problem.

As a workaround you could use un-minified version of Highstock that has [correct .d.ts](https://unpkg.com/highcharts@7.0.1/highstock.src.d.ts)

@bre1470 Looks like a [single typo](https://github.com/highcharts/highcharts-declarations-generator/commit/f0f2ca0d69291ee8ab26026a55f5e2d3c223aba7#diff-0a9374812112da7378eb2c1a1015af3fR6).

**Comment by bre1470** [comment]
I am sorry for this. The bug actually should have been detected by dtslint here: https://github.com/highcharts/highcharts/blob/master/test/typescript/highstock.ts#L17 , because the stockChart factory is only available, if the stock module declaration can be found.

We will now improve our test systems, so it detects bugs in general more reliable.

**Comment by scott-ho** [comment]
Hope this could be fixed soon.

**Comment by SayakMukhopadhyay** [comment]
Is there a timeline for this fix to be released?

**Comment by KacperMadej** [comment]
The fix is expected to be published within next few weeks. It should be a part of next path release (7.0.2 or any next version of Highcharts that will be published) if a new version of declarations generator (probably 1.0.6) will be included there with the typo fix (at this very moment the fix is a part of 1.0.6 version of the declarations generator).

tl;dr ETA within next few weeks.

**Comment by bre1470** [comment]
This issue is now fixed with the master branch and will be part of the next release, due within a few weeks.

**Comment by slavede** [comment]
How much is a few? The thing I'm wondering, why can't you bump minor version to 7.0.**2** and release that (or even 7.1.0)? And if you have one additional major release coming that's fine, you can bump it again later?

I'm not trying to be ungrateful here, but to me it seems really critical issue to leave it hanging for weeks just because it doesn't match release plan.

**Comment by bre1470** [comment]
We prepare v7.0.2 for release in the next ten days.

**Comment by cebor** [comment]
When you impot highstock from the src files there is not this issue:

```ts
import * as Highstock from 'highcharts/highstock.src';
```

**Comment by bre1470** [comment]
Thank you for your feedback, @chukwu. We are sorry for the frustration this bug is causing.

The workarounds mentioned by @KacperMadej and @cebor (Thank you!) are helpful temporary solutions during the development process, which you should consider:
```
import * as Highcharts from 'highcharts';
import StockModule from 'highcharts/modules/stock';
StockModule(Highcharts);
```
- or -
```
import * as Highstock from 'highcharts/highstock.src';
```

**Comment by chukwu** [comment]
The main issue for me is with angular-highchart plugin/wrapper. The plugin doesn't work with highchart >7 as pointed in https://github.com/cebor/angular-highcharts/issues/250
Plugin developer can't do anything until highchart is fixed. So, I'm forced to install highchart 6.2--

**Comment by bre1470** [comment]
Just to make sure, that everyone gets notified: The v7.0.2 update has been released and fixes this issue.

**Comment by attilacsanyi** [comment]
**Thanks for the quick release Highcharts team! Really appreciated :)**
